src/backend/distributed/executor/partitioned_intermediate_results.c

```diff
@@ -540,6 +541,12 @@ PartitionedResultDestReceiverShutdown(DestReceiver *dest)
 		DestReceiver *partitionDest = self->partitionDestReceivers[i];
 		partitionDest->rShutdown(partitionDest);
 	}
+	i = -1;
+	while ((i = bms_next_member(self->startedDestReceivers, i)) >= 0)
```

Comment:
seems like this logic is already in PartitionedResultDestReceiverDestroy? should we call that instead?

Reply:
Here this function is setting the variable "startedDestReceivers" to NULL just after the for loop..
So the PartitionedResultDestReceiverDestroy will not be able to call the "destroy" call function registered from the worker_shard_copy.c.. Those function are freeing some memory allocated, so if they are not called, it may result in some memory leak I think..